services/auth/src/signaturevalidator.js

```diff
@@ -291,8 +293,8 @@ const createSignatureValidator = function(options) {
 
         credentials = authResult.credentials;
         attributes = authResult.artifacts; // Hawk uses "artifacts" and "attributes"
-      } else {
-        // If there is no authorization header we'll attempt a login with bewit
+      } else if (/^\/.*[\?&]bewit\=/.test(req.resource)) {
```

Comment:
just curious why using a regex instead of `req.query`? Have we disabled query parsing maybe? Let's comment if so.

Reply:
it looks like it. logging `req` shows:

```
{
  method: 'GET',
  resource: '/?bewit=...',
  host: 'test.taskcluster.net',
  port: 443,
  sourceIp: '127.0.0.1'
}
```

i'll put a comment.

Reply:
This regexp is, I think, copied and reduced from one a few lines later:

https://github.com/taskcluster/taskcluster/blob/f79e42cd338091b677292855bb0ada9fe5e1ac87/services/auth/src/signaturevalidator.js#L304-L307

I think that code parses `req.resource` rather than `req.query` in an effort to do precisely the same thing that the hawk library does.  Switching that around would run the risk of changing something with URL escaping, for example.

In the version before this PR, we just called `hawk.uri.authenticate` whenever there was no authorization header, either ending up with scopes or no-auth.  Now, we need to distinguish those two cases into three, so we need to see if there's a `bewit` query arg.  I'm guessing someone copy-pasted and modified the regexp to do so.  I'm don't totally understand why we need to distinguish three cases, but assuming that's required, this way of doing so makes sense in context.

Reply:
Sounds good to both of you. I'm convinced this one is not an issue!